airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -199,8 +199,6 @@ def _read_credentials_from_connection(self) -> Tuple[Optional[str], Optional[str
                 self.extra_config.get("profile"),
             )
             self.log.info("Credentials retrieved from extra_config['s3_config_file']")
-        else:
-            self.log.info("No credentials retrieved from Connection")
```

Comment:
Personally I’d appreciate having this log since otherwise it’s more difficult to figure out why something _didn’t_ happen.

Reply:
Hmm, interesting. The problem with this log line is that it's one of the worst offenders for log spam, since it sits in an else that's guaranteed to run even if you're not trying to use a connection for credentials at all (see the example screenshot I've attached in the conversation tab). Which as far as negative case logging goes, isn't that useful IMHO. This log line also adds no additional runtime context (it logs no runtime data that's useful for debugging) and it can simply be inferred if you don't see any of the other log lines from above.

However, if all that is still not convincing, then perhaps we can agree to add it back, but at debug level?

Let me know what you think!

Reply:
Would it be possible to add a log only when we know we’re trying to get a credential? Maybe log somewhere else?

Reply:
Hey,

There is perhaps a way to modify the logic and the way the conditions are constructed to reduce the scope of the negative case but nothing jumps out at me as being trivial. I really would rather not modify functional code in this PR as it makes it much more risky to cause regressions for the simple goal of log level clean up.

I really do think everything you need is already there: If you're trying to use a connection (in any of the 3 ways above), you'll get a log line printed telling you that you are doing so. If you're trying to use a connection and you don't see such a log being printed, something has gone wrong and you should re-asses. The else logging adds no additional data.

Does adding back the else condition at debug level as a compromise not satisfy your particular use case?